docs/SCTValidation.md

```diff
@@ -0,0 +1,259 @@
+ - The CA signs the resulting precertificate using the pre-issuer's key.
+ - The submits the whole chain (precert, pre-issuer, true-issuer, ... root) to
+   the Log.
+ - The Log removes the poison and extracts the inner `tbsCertificate`.
+ - The Log notices that the direct issuer has the Certificate Transparency
+   extended key usage, so treats the next entry in the chain as the true issuer.
+ - The Log updates the Issuer and (optional) Authority Key Identifier extension
+   to match the true issuer rather than the pre-issuer.
+ - The Log combines the resulting inner `tbsCertificate` with the hash of the
+   (true) issuer's key, to form the core `PreCert` data that the Log deals with.
+ - The Log sends an SCT back to the CA, which includes a timestamp as usual but
+   has a signature over data including the `PreCert` bundle of `tbsCertificate`
+   and issuer key hash.
+ - The CA builds an SCT list extension that includes this SCT, attaches it to
+   the original (un-poisoned) certificate, and signs the whole thing with the
+   true issuer's key.
+
+This is complicated, but it's worth pointing out that only the CA and the Log
+need to deal with all of this; the SCT (and the tree leaf it corresponds to)
+only covers data that pertains to the final certificate.  This means that a
+client who receives a certificate with an embedded SCT can just do the
+[same checks](#embedded-scts) as for any other embedded SCT.
+
+
+## SCT Validation Steps
+
+There are two main aspects to validating an SCT:
+ - Signature Validation: this requires the following (in addition to the SCT itself):
+     - the Log's public key
+     - the [pre]certificate data that the signature encompasses
+     - (for a precertificate) the issuer's public key
+ - Inclusion Checking: this requires the following (in addition to the SCT
+   itself):
+     - enough time to have passed (the MMD) for the certificate to be incorporated
+     - the Log's URL (for accessing the `get-proof-by-hash`
+       [entrypoint](https://tools.ietf.org/html/rfc6962#section-4.5))
+     - the [pre]certificate data that the signature encompasses
+     - (for a precertificate) the issuer's public key
+     - a published tree size for the Log (via the `get-sth`
+       [entrypoint](https://tools.ietf.org/html/rfc6962#section-4.1)).
+
+
+### Embedded SCTs
+
+To check SCTs that are embedded in an X.509 certificate, a client needs to
+rebuild the [precertificate](#precertificates) data that the SCT and the leaf
+hash encompasses:
+
+ - The SCT list extension must be removed.
+ - The inner `tbsCertificate` data must be extracted, and combined with the hash
+   of the issuer's public key,
+
+### Inclusion Checking
+
+To perform on-line inclusion checking for an SCT, the client needs to generate
+the *leaf hash* for the submitted entry, to submit in the `get-proof-by-hash`
+[entrypoint](https://tools.ietf.org/html/rfc6962#section-4.5).
+
+This is the SHA-256 hash of a zero byte followed by the TLS encoding of a
```

Comment:
A down-to-the-bytes worked example would help here too, I think.

Reply:
As above.